config/transformers/namePathToKebabCase.ts

```diff
@@ -8,5 +8,12 @@ import StyleDictionary from 'style-dictionary'
  */
 export const namePathToKebabCase: StyleDictionary.Transform = {
   type: `name`,
-  transformer: (token: StyleDictionary.TransformedToken) => token.path.join('-')
+  transformer: (token: StyleDictionary.TransformedToken, options?: StyleDictionary.Platform): string => {
+    return (
+      [options?.prefix, ...token.path]
+        // remove undefined if exists
+        .filter((part: unknown): part is string => typeof part === 'string')
```

Comment:
do you need to narrow type with `is` here? what happens if you just put part:string? or better yet, infer type from options?

Reply:
I do, yes. Otherwise I get a ts error as ts does not understand that the filter will only allow strings to pass through.
If options is not defined the array has an undefined value in it, right?

I had tried `<string | undefined>[options?.prefix, ...token.path]` and even `<string>[options?.prefix, ...token.path]` but both gave me errors.

Is there a problem with `part is string` in general?

Reply:
not a problem with it, I was just reading it and confused why we needed unknown, which feels a bit icky.. and assumed type inference would just work 🤷‍♂️ . Carry on!